packages/stargate/src/stargateclient.spec.ts

```diff
@@ -117,6 +125,43 @@ describe("StargateClient", () => {
     });
   });
 
+  describe("getBlock", () => {
+    it("works for latest block", async () => {
+      pendingWithoutSimapp();
+      const client = await StargateClient.connect(simapp.tendermintUrl);
+      const response = await client.getBlock();
+
+      expect(response.id).toMatch(tendermintIdMatcher);
+
+      expect(response.header.height).toBeGreaterThanOrEqual(1);
+      expect(response.header.chainId).toEqual(await client.getChainId());
+      expect(new ReadonlyDate(response.header.time).getTime()).toBeLessThan(ReadonlyDate.now());
+      expect(new ReadonlyDate(response.header.time).getTime()).toBeGreaterThanOrEqual(
+        ReadonlyDate.now() - 5_000,
+      );
+
+      expect(Array.isArray(response.txs)).toEqual(true);
+    });
+
+    it("works for block by height", async () => {
+      pendingWithoutSimapp();
+      const client = await StargateClient.connect(simapp.tendermintUrl);
+      const height = (await client.getBlock()).header.height;
+      const response = await client.getBlock(height - 1);
+
+      expect(response.id).toMatch(tendermintIdMatcher);
+
+      expect(response.header.height).toEqual(height - 1);
+      expect(response.header.chainId).toEqual(await client.getChainId());
+      expect(new ReadonlyDate(response.header.time).getTime()).toBeLessThan(ReadonlyDate.now());
+      expect(new ReadonlyDate(response.header.time).getTime()).toBeGreaterThanOrEqual(
+        ReadonlyDate.now() - 5_000,
+      );
+
+      expect(Array.isArray(response.txs)).toEqual(true);
```

Comment:
Jasmine can do this as well `expect(response.txs).toEqual(jasmine.any(Array)`, which allows you to cobine a lot of stuff like:

```ts
expect(response).toEqual(jasmine.objectContaining({
  header: {
    height: height - 1,
    chainId: await client.getChainId(),
  },
  txs: jasmine.any(Array)
});
// time needs to be tested separately
```